src/main/java/com/example/Restaurant_Management_System_REST_API/service/InventoryItemService.java

```diff
@@ -36,25 +36,27 @@ public InventoryItemDTOResponse create(InventoryItemDTORequest inventoryItemDTOR
         InventoryItem inventoryItem = modelMapper.map(inventoryItemDTORequest, InventoryItem.class);
 
         //Checking if provided inventory item with the same supplier already exists in database
-        checkIfThisInventoryItemWithThisSupplierAlreadyExists(inventoryItemDTORequest);
+        if(inventoryItemDTORequest.getSupplier() != null)
+            checkIfThisInventoryItemWithThisSupplierAlreadyExists(inventoryItemDTORequest);
 
         inventoryItem.setSupplier(supplier);
         inventoryItemRepository.save(inventoryItem);
 
         return modelMapper.map(inventoryItem, InventoryItemDTOResponse.class);
     }
 
-    private void checkIfThisInventoryItemWithThisSupplierAlreadyExists(InventoryItemDTORequest inventoryItemDTORequest) throws ObjectAlreadyExistException {
-        Optional<InventoryItem> optionalInventoryItem = inventoryItemRepository.findByName(inventoryItemDTORequest.getName());
+    private void checkIfThisInventoryItemWithThisSupplierAlreadyExists(InventoryItemDTORequest inventoryItemDTORequest)
```

Comment:
Why such complicated name? It's very long and not descriptive.

Reply:
and you see for me it is exact and very descriptive - therefore it is so long -> this methods checks if such InventoryItem with specific Supplier exists that is why I added 'This' twice - to highlight it.
anyways
what I can propose instead?
maybe: checkInventoryItemWithSupplierExists() ?

Reply:
Look, @BartoliniAndBorderCollies, long names doesn't improve readability. You have comments available to describe the code. Especially, when you can't think of a better name for a method.

I'd suggest using:
`assertUniqueInventorySupplier`
`verifyItemWithSupplierExists`
`doesItemWithSupplierExist`
`confirmItemSupplierUniqueness`

or a more custom ones:
`validateItemSupplierCombination`
`checkForDuplicateItemSupplierPair`
`ensureItemAndSupplierUniqueness`

Your case here is so specific that's the best way to approach is to get method a proper documentation and a simple name.

Reply:
got it!